### PR TITLE
Make F11 work for fullscreen of the widget

### PR DESCRIPTION
**Pull request by MTRNord**
**:heavy_check_mark: Checklist**

- [ ] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [ ] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: packages/react-sdk/src/components/Layout/useFullscreenMode.ts [hunk 1]
```diff
@@ -64,6 +64,21 @@ export function useFullscreenMode(): UseFullScreenModeResult {
     };
   }, [setFullscreenModeState]);
 
+  useEffect(() => {
+    function listener(e: KeyboardEvent) {
```
Comment (weeman1337) [inline packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: We use [React Hotkeys Hook](https://react-hotkeys-hook.vercel.app/) in other places. It should give us more control over hotkeys in different context (such as only enable them in specific places).

Can you switch to it?

**Review by weeman1337 — CHANGES_REQUESTED** [review]

File: packages/react-sdk/src/components/Layout/useFullscreenMode.ts (hunk 1 above)
Comment (MTRNord) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: @weeman1337 I tried but it seems like either it works 2 times and then gets stuck in `TypeError: Fullscreen request denied` in firefox or changing to keyup makes it work once and then not trigger again. So something seems wonky with this hook which makes it unreliable compared to using a classic keyhandler

Comment (MTRNord) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: The hook seems to cause mainly `Anfrage für Vollbildmodus wurde abgelehnt, weil Element.mozRequestFullScreen() nicht aus einem kurz laufenden Benutzer-generierten Ereignis-Handler aufgerufen wurde.`

Comment (MTRNord) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: It seems like it works when I actively click on the board in between but it doesn't just with the keybind despite https://developer.mozilla.org/en-US/docs/Web/Security/User_activation claiming keydown is accepted. I can only assume that firefox fails due to some indirection or memo thing here

**Review by HarHarLinks — COMMENTED** [review]
It works. I don't have anything to add about the technical discussion in the other thread in either direction. I am not sure if the implemented behavior is required or desirable.

File: packages/react-sdk/src/components/Layout/useFullscreenMode.ts (hunk 1 above)
Comment (weeman1337) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: Then lets keep it like it. Adding a comment

Comment (weeman1337) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: Also getting this without the hotkeys hook… Seems like it is triggered twice.

Comment (weeman1337) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: I think this can be caused by Reacts strict mode.

Comment (weeman1337) [reply packages/react-sdk/src/components/Layout/useFullscreenMode.ts:68]: Or the hook being used in two places.

**Comment by weeman1337** [comment]
I did update the PR:
- It now has error handling
- The hook is used in one place only (and it should really only used from layout state)
- It uses react hotkeys
- It remembers the internal state, whether there is already an ongoing full-screen request 

With that changes, it works on my machine™

@HarHarLinks can you re-test it? Also in different browsers.

**Review by weeman1337 — APPROVED** [review]

**Comment by HarHarLinks** [comment]
I can do that once there is a product decision whether this is actually desireable behavior.

**Comment by HarHarLinks** [comment]
This PR changes behavior such that pressing F11 may switch it into fullscreen mode. This has the following issues:

Because Element Web/Desktop, as the currently only client really implementing widgets, hosts its widgets in iframes, it causes inconsistent behavior in browsers. Depending on the current focus, a different part of the app will capture the keyboard shortcut. If the user has most recently used the Element UI e.g. to send a message or navigate to a room, or interacted with another widget (e.g. in the NeoToolSuite), then the focus is outside the NeoBoard and pressing F11 will maximize Element or potentially the other widget instead of NeoBoard. Only when the last interaction was in NeoBoard, then this will work as a user might expect. Similar issues arise for other apps using the NeoBoard react-sdk, though the situation may be more or less severe than in the case of the widget.

Further, F11 is one of multiple de-facto standards on the web that puts the page into fullscreen, hiding the tab and address bars etc. It does not put some selected part of the page only into fullscreen. The proposal is breaking with that convention.

For comparison of purely the UX component: if I'm watching a video on YouTube, regardless where if I've last clicked on the player or the page, F11 always maximizes the page, not the player. (The custom shortcut <kbd>f</kbd> in the meantime is subject to the technical limitations at play here.)

Therefore, we reject this for now.

**Pull request closed without merging**